src/mlpack/methods/adaboost/adaboost_fit_main.cpp

```diff
@@ -0,0 +1,180 @@
+ *   url = {http://dx.doi.org/10.1023/A:1007614523901},
+ *   doi = {10.1023/A:1007614523901},
+ *   acmid = {337870},
+ *   publisher = {Kluwer Academic Publishers},
+ *   address = {Hingham, MA, USA},
+ *   keywords = {boosting algorithms, decision trees, multiclass classification,
+ *   output coding}
+ * @endcode
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include <mlpack/prereqs.hpp>
+#include <mlpack/core/util/io.hpp>
+
+#ifdef BINDING_NAME
+  #undef BINDING_NAME
+#endif
+#define BINDING_NAME adaboost_fit
+
+#include <mlpack/core/util/mlpack_main.hpp>
+#include <mlpack/core/data/normalize_labels.hpp>
+#include "adaboost.hpp"
+#include "adaboost_model.hpp"
+
+using namespace mlpack;
+using namespace std;
+using namespace arma;
+using namespace mlpack::adaboost;
+using namespace mlpack::tree;
+using namespace mlpack::perceptron;
+using namespace mlpack::util;
+
+// Program Name.
+BINDING_USER_NAME("AdaBoost Training");
+
+// Short description.
+BINDING_SHORT_DESC(
+    "Training AdaBoost model.");
+
+// Long description.
+BINDING_LONG_DESC("");
+
+// Example.
+BINDING_EXAMPLE(
+    IMPORT_EXT_LIB() + "\n" +
+    IMPORT_SPLIT() + "\n" +
+    IMPORT_THIS("adaboost") + "\n" +
+    GET_DATASET("X", "https://example.com") + "\n" +
+    GET_DATASET("y", "https://example.com") + "\n" +
+    SPLIT_TRAIN_TEST("X", "y", "X_train", "y_train", "X_test", "y_test", 
+    "0.2") + "\n" +
+    CREATE_OBJECT("model", "adaboost") + "\n" +
+    CALL_METHOD("model", "fit", "training", "X_train", "labels", "y_train"));
+
+// See also...
+BINDING_SEE_ALSO("AdaBoost on Wikipedia", "https://en.wikipedia.org/wiki/"
+        "AdaBoost");
+BINDING_SEE_ALSO("Improved boosting algorithms using confidence-rated "
+        "predictions (pdf)", "http://rob.schapire.net/papers/SchapireSi98.pdf");
+BINDING_SEE_ALSO("Perceptron", "#perceptron");
+BINDING_SEE_ALSO("Decision Stump", "#decision_stump");
+BINDING_SEE_ALSO("mlpack::adaboost::AdaBoost C++ class documentation",
+        "@doxygen/classmlpack_1_1adaboost_1_1AdaBoost.html");
+
+// Input for training.
+PARAM_MATRIX_IN("training", "Dataset for training AdaBoost.", "t");
+PARAM_UROW_IN("labels", "Labels for the training set.", "l");
+
+// Training options.
+PARAM_INT_IN("iterations", "The maximum number of boosting iterations to be run"
+    " (0 will run until convergence.)", "i", 1000);
+PARAM_DOUBLE_IN("tolerance", "The tolerance for change in values of the "
+    "weighted error during training.", "e", 1e-10);
+PARAM_STRING_IN("weak_learner", "The type of weak learner to use: "
+    "'decision_stump', or 'perceptron'.", "w", "decision_stump");
+
+// Loading/saving of a model.
+// PARAM_MODEL_IN(AdaBoostModel, "input_model", "Input AdaBoost model.", "m");
```

Comment:
```suggestion
// Allow saving the model.
```
I think we can update this comment now. :+1: